File: Knjigoteka.Model/Entities/Borrowing.cs
using Microsoft.Identity.Client;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Knjigoteka.Model.Entities
{
    public class Borrowing
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int BookId { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public int BranchId { get; set; }
        public int? ReservationId { get; set; }
        [Required]
        public DateTime BorrowedAt { get; set; } = DateTime.Now;
        [Required]
        public DateTime DueDate { get; set; } = DateTime.Now.AddDays(30);
        public DateTime? ReturnedAt { get; set; }
        public bool IsLate => ReturnedAt.HasValue && ReturnedAt.Value > DueDate;
        [ForeignKey(nameof(BookId))]
        public Book Book { get; set; } = null!;
        [ForeignKey(nameof(UserId))]
        public User User { get; set; } = null!;
        [ForeignKey(nameof(BranchId))]
        public Branch Branch { get; set; } = null!;
        [ForeignKey(nameof(ReservationId))]
        public Reservation? Reservation { get; set; }
    }
}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/borrowing.dart
class Borrowing {
  final int id;
  final int bookId;
  final String bookTitle;
  final int userId;
  final String userName;
  final int branchId;
  final String branchName;
  final int? reservationId;
  final DateTime borrowedAt;
  final DateTime dueDate;
  final DateTime? returnedAt;
  final bool isLate;

  Borrowing({
    required this.id,
    required this.bookId,
    required this.bookTitle,
    required this.userId,
    required this.userName,
    required this.branchId,
    required this.branchName,
    this.reservationId,
    required this.borrowedAt,
    required this.dueDate,
    this.returnedAt,
    required this.isLate,
  });

  factory Borrowing.fromJson(Map<String, dynamic> json) {
    return Borrowing(
      id: json['id'],
      bookId: json['bookId'],
      bookTitle: json['bookTitle'] ?? '',
      userId: json['userId'],
      userName: json['userFullName'] ?? '',
      branchId: json['branchId'],
      branchName: json['branchName'] ?? '',
      reservationId: json['reservationId'],
      borrowedAt: DateTime.parse(json['borrowedAt']),
      dueDate: DateTime.parse(json['dueDate']),
      returnedAt: json['returnedAt'] != null
          ? DateTime.parse(json['returnedAt'])
          : null,
      isLate: json['isLate'] ?? false,
    );
  }
}
